deployments/utoronto/image/Dockerfile

```diff
@@ -108,16 +108,6 @@ RUN apt-get update -qq --yes && \
         libgsl0-dev \
         > /dev/null
 
-# See https://github.com/utoronto-2i2c/jupyterhub-deploy/pull/101
-RUN apt-get update -qq --yes && \
-    apt-get install --yes -qq \
-        libudunits2-dev \
-        libgdal-dev \
-        gdal-bin \
-        libgeos-dev \
-        libproj-dev \
-        > /dev/null
-
```

Comment:
These requirements are listed on `sf`'s package manager page  [here](https://packagemanager.rstudio.com/client/#/repos/1/packages/sf). (Some of them are listed as system requirements for other of these new packages too).

However, this caused the following error:
```
checking GDAL: checking whether PROJ is available fur running:... ./gdal_proj: error while loading shared libraries: libgdal.so.29: cannot open shared object file: No such file or directory
no
configure: error: OGRCoordinateTransformation() does not return a coord.trans: PROJ not available?
ERROR: configuration failed for package ‘sf’
```

Mainly because we're already installing a prod version of these pkgs and we ended up having diffrenet versions of GDAL present https://github.com/utoronto-2i2c/jupyterhub-deploy/blob/e1fd790000734f1356a5a78357ccca0df0d21f38/deployments/utoronto/image/Dockerfile#L144-L146

Reply:
Nope, this didn't solve the issue and was a pretty stupid reasoning :(

 We need the `-dev` version of the pkgs, because otherwise the dynamic lib `libgdal.so` doesn't get installed at all.
 
 So it's probably a combined issue of:
- how to tell `sf` where the lib is :( 
- `libgdal26` comes with `GDAL` version 3.2.2 and `libgdal-dev` comes with `GDAL` version 3.3.1

Reply:
Hmm, maybe install just the `dev` version of the pkgs? @yuvipanda, do you know if there are pkgs that have strong dependencies on the versions installed here:
https://github.com/utoronto-2i2c/jupyterhub-deploy/blob/e1fd790000734f1356a5a78357ccca0df0d21f38/deployments/utoronto/image/Dockerfile#L138-L147

Reply:
So, `sf` wants  libgdal.so.29, but `/usr/lib` has only:
- libgdal.so
- libgdal.so.26
- libgdal.so.26.0.4

On apt, the latest available on is 28: https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=libgdal&searchon=names, which makes me think that we might need to install from [ubuntugis-unstable PPA](https://github.com/r-spatial/sf#ubuntu). Or we could create a symlink from `libgdal.so` to `libgdal.so.29` and hope it works :shrug:

Reply:
Aaah, what a fun mess :)

The way I normally debug this is to get the image to build by removing the problematic packages, and then play around with different `apt-get` commands and R package installation commands until things work.

Here's some suggestions to try:

1. Check what version of R is installed, and see if we need to upgrade that.
2.. Upgrade all other R package versions, maybe this is causing a conflict that causes issues?

Just googling for `configure: error: OGRCoordinateTransformation() does not return a coord.trans: PROJ not available?` also brought in a ton of other solutions - any luck with those?